hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/upgrade/OMPrepareRequest.java

```diff
@@ -212,12 +212,6 @@ private static long waitForLogIndex(long minOMDBFlushIndex,
               "OM database flushed index %d which is less than the minimum " +
               "required index %d.",
           flushTimeout.getSeconds(), lastOMDBFlushIndex, minOMDBFlushIndex));
-    } else if (!ratisStateMachineApplied) {
-      throw new IOException(String.format("After waiting for %d seconds, " +
-              "Ratis state machine applied index %d which is less than" +
-              " the minimum required index %d.",
-          flushTimeout.getSeconds(), lastRatisCommitIndex,
-          minRatisStateMachineIndex));
```

Comment:
We should remove `minRatisStateMachineIndex` and `ratisStateMachineApplied` from the other parts of this function as well if we are no longer using them. We also need to make sure that `om.getRatisSnapshotIndex()` returning an index means that was actually written to RocksDB. Checking the transaction info table may be a better option.